Sources/NIOSSL/SSLConnection.swift

```diff
@@ -407,6 +407,24 @@ internal final class SSLConnection {
     func extractUnconsumedData() -> ByteBuffer? {
         return self.bio?.evacuateInboundData()
     }
+    
+    /// Retries the `TLSVersion` used on a `Channel` through the `NIOSSLHandler` APIs.
+    /// Note that the .tlsv12 has been added to the default cases as this is what BoringSSL sets as the default anyways.
+    func getTLSVersionForConnection() -> TLSVersion {
+        let uint16Version = CNIOBoringSSL_SSL_version(self.ssl)
+        switch uint16Version {
+          case TLS1_3_VERSION:
+            return .tlsv13
+          case TLS1_2_VERSION:
+            return .tlsv12
+          case TLS1_1_VERSION:
+            return .tlsv11
+          case TLS1_VERSION:
+            return .tlsv1
+        default:
+            return .tlsv12 // This is the default for BoringSSL
```

Comment:
This should probably return `nil` if we don't have an answer.

Reply:
I refactored this not to use `.tlsv12` in the latest and to return `nil` instead on the default case.  This of course changes `TLSVersion` to be an optional returned here but that didn't have too much impact on the final usage of this API.  Let me know your thoughts.